app/tests/api/inventory.test.ts

```diff
@@ -145,4 +167,102 @@ describe("Inventory API", () => {
     });
     assert.equal(res.status, 404);
   });
+
+  it("should calculate progress for an inventory", async () => {
+    // setup mock data
+    const existingInventory = await db.models.Inventory.findOne({
+      where: { year: inventory.year },
+    });
+    assert.notEqual(existingInventory, null);
+    const sectorNames = ["TEST1", "TEST2", "TEST3"];
+    const userSource = await db.models.DataSource.create({
+      datasourceId: randomUUID(),
+      sourceType: "user",
+      name: "XX_INVENTORY_TEST_USER",
+    });
+    const thirdPartySource = await db.models.DataSource.create({
+      datasourceId: randomUUID(),
+      sourceType: "third_party",
+      name: "XX_INVENTORY_TEST_THIRD_PARTY",
+    });
+    const sources = [userSource, thirdPartySource, null];
+
+    for (const sectorName of sectorNames) {
+      const sectorId = randomUUID();
+      await db.models.Sector.create({
+        sectorId,
+        sectorName: "XX_INVENTORY_" + sectorName,
+      });
+      const sectorValueId = randomUUID();
+      await db.models.SectorValue.create({
+        sectorValueId,
+        sectorId,
+        inventoryId: existingInventory!.inventoryId,
+      });
+      for (let i = 0; i < sectorNames.length; i++) {
+        const subsectorId = randomUUID();
+        await db.models.SubSector.create({
+          subsectorId,
+          sectorId,
+          subsectorName: "XX_INVENTORY_" + sectorName + "_" + sectorNames[i],
+        });
+        if (sources[i] != null) {
+          await db.models.SubSectorValue.create({
+            subsectorValueId: randomUUID(),
+            subsectorId,
+            sectorValueId,
+            datasourceId: sources[i]?.datasourceId,
+            inventoryId: existingInventory!.inventoryId,
+          });
+        }
+      }
+    }
+
+    const url = `http://localhost:3000/api/v0/city/${locode}/inventory/${inventory.year}/progress`;
```

Comment:
Should this be just /api/v0/.....?

Reply:
So this needs to be a fully qualified URL in order to successfully pass the URL parser.
The thing is it's not actually used anywhere, so we might as well pass a placeholder URL in for all routes.
I just did this mainly for documentation in the tests, so it's clear which route is used.

Reply:
The URL never actually gets fetched as I call the endpoint handler directly to make the tests faster.